lib/registry.js

```diff
@@ -26,42 +26,51 @@ class Registry {
 	}
 
 	getMetricAsPrometheusString(metric, conf) {
-		const opts = Object.assign({}, defaultMetricsOpts, conf);
```

Comment:
The point with using Object.assign like this was both a good way to extend and overwrite the default settings but also a way to make sure a specific configuration value always was there. I know that you changed on how you call this method from the metrics function but this function is public available so it is a breaking change.

Reply:
I can revert this change, not the biggest perf win. Good call on the public api change potential.